### ISSUE TITLE
[Feature Request] Provides a better common logger

### ISSUE THREAD

**Issue by Jeffwan** [kubeflow/common#60]
Currently, `kubeflow/common` provides a logger utils for operator to use.  It uses `github.com/sirupsen/logrus` underneath. 

https://github.com/kubeflow/common/blob/master/pkg/util/logger.go#L20

The problem I notice is new operators using `kubebuilder` have inbuilt logger for operator logics.
This makes the logs really messy. 

Can we provide a interface layer like [logr](https://github.com/go-logr/logr) and provide method like `SetLogger` and different operator can use their own implementation?

**Comment by terrytangyuan** [comment]
Agreed. Having a unified interface would make it easier for downstream apps to consume the logs.

**Comment by terrytangyuan** [comment]
cc @ShuhanYan @carmark in case you are interested
